queues/queues.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2021, Aryan Ahadinia. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package queues provides an abstract Queue interface.
+//
+// In computer science, a queue is a collection of entities that are maintained in a sequence and can be modified by the addition of entities at one end of the sequence and the removal of entities from the other end of the sequence. By convention, the end of the sequence at which elements are added is called the back, tail, or rear of the queue, and the end at which elements are removed is called the head or front of the queue, analogously to the words used when people line up to wait for goods or services.
+// The operation of adding an element to the rear of the queue is known as enqueue, and the operation of removing an element from the front is known as dequeue. Other operations may also be allowed, often including a peek or front operation that returns the value of the next element to be dequeued without remove it.
+//
+// Reference: https://en.wikipedia.org/wiki/Queue_(abstract_data_type)
+package queues
+
+import "github.com/emirpasic/gods/containers"
+
+// Queue interface that all queues implement
+type Queue interface {
+	Enqueue(value interface{})
```

Comment:
What about the priority queue here in the future which requires priority in addition to the value when inserting?

https://en.wikipedia.org/wiki/Priority_queue

Also consider (future-proofing a bit) the circular buffer implementation, it would not have the enqueue/dequeue functions neither.